src/mtype.c

```diff
@@ -5635,8 +5635,8 @@ Type *TypeFunction::semantic(Loc loc, Scope *sc)
                 }
                 else
                 {
-                    error(loc, "auto can only be used for template function parameters");
-                    errors = true;
+                    fparam->storageClass &= ~STCauto;
+                    fparam->storageClass |= STCrvref;
```

Comment:
Code is LGTM.

My proposal is: `'auto' can only be used along with 'ref' for function parameters`

I feel that `auto` is not an option for the ref parameter, rather `auto ref` is a kind of *combined attribute*. So the message suggests to combine `auto` and `ref`.